### ISSUE TITLE
Centralization theoretical max appears to be calculated incorrectly

### ISSUE THREAD

**Issue by jdfoote** [igraph/rigraph#369]
When calculating degree centralization, a star graph should be the most centralized (i.e., centralization should equal 1).

However, the following code results in a centralization of 0.75.
```
x = make_star(8, mode = 'undirected')
plot(x)
centralization.degree(x)$centralization
```

This bug was identified on [SO](https://stackoverflow.com/questions/39236981) years ago suggesting an off-by-one error but it appears that an issue was never created.

This appears to be correct, as `centralization.degree.tmax(x) != centralization.degree(x)$theoretical_max`

- [X] This issue is for the C core of igraph.
- [X] This issue is a bug report or a feature request, not a support question.

**Comment by szhorvat** [comment]
Strangely, I could reproduce this in the Mathematica interface as well.

**Comment by szhorvat** [comment]
OK, this is not a bug. It happens because self-loops are considered by default. 

Use `centr_degree(g, loops=F)`

**Comment by jdfoote** [comment]
Interesting. 

It seems like this should be consistent between `centralization.degree.tmax()` and `centralization.degree`.

It seems like most of the graph generation functions default to not producing self loops. Maybe this would be better to default to no self-loops?

If self loops exist in the graph, then an exception could be raised (or a message passed that the calculation had changed to include self loops)?

**Comment by torfason** [comment]
Agreed, consistency across functions and general expectations regarding degree centralization suggest that the default should be `loops=FALSE`. I think it is unlikely that anyone who really is intending to use `loops=TRUE` is actually relying on this default. Much more likely that people who did not think of loops are now getting an incorrect result because the did not realize how the default works.

But I agree with @jdfoote that it might be helpful to throw a warning if `centr_degree()` is called on graphs that contain loops without explicitly setting the `loops` argument.

**Comment by szhorvat** [comment]
Adding to 1.3 milestone because we need to make a decision about which way to go here, and do the appropriate deprecations.

I don't have a good suggestion here. The most reasonable thing is indeed to make specifying `loops` mandatory.

**Comment by torfason** [comment]
If the idea would be to reduce the disruption due to unexpected breakage on upgrade, one approach would be to add a `warning` in one version if `loops` were not specified, and then make it an `error` in the version after that.

**Comment by ntamas** [comment]
Yes, that's what I am going to do. igraph 1.3.0 will warn the user if `loops` is not specified, and we make it mandatory in 1.4.0.

**Comment by ntamas** [comment]
Fixed in 63c537d8
